File: game/forest/hurt_trap_talking.py
import pygame

from animation import Animation
from audio_helper import AudioHelper
from forest.forest_resources import ForestResources
from forest.forest_utils import ForestUtils
from state import State


class HurtTrapTalking(State):
    def process_events(self, events):
        if self.get_frame_index() >= len(ForestResources.sync_trap):
            return ForestUtils.reduce_lives(self.context)

        return None

    def on_enter(self) -> None:
        super().on_enter()
        AudioHelper.play(ForestResources.speak_trap, self.context.country)

    def render(self, screen):
        screen.blit(Animation.get_sync_frame(ForestResources.hugo_traptalk, ForestResources.sync_trap, self.get_frame_index()), (0,0))

File: game-web/src/forest/hurtTrapTalking.ts
/** Port of game/forest/hurt_trap_talking.py */

import { State, type StateClass } from '../state';
import { Animation } from '../animation';
import { AudioHelper } from '../audioHelper';
import { ForestResources } from './forestResources';
import { ForestUtils } from './forestUtils';
import type { PhoneEvents } from '../phoneEvents';

export class HurtTrapTalking extends State {
  processEvents(_events: PhoneEvents): StateClass | null {
    if (this.getFrameIndex() >= ForestResources.syncTrap.length) {
      return ForestUtils.reduceLives(this.context, 2); // Trap
    }
    return null;
  }

  onEnter(): void {
    super.onEnter();
    AudioHelper.play(ForestResources.speakTrap);
  }

  render(ctx: OffscreenCanvasRenderingContext2D): void {
    ctx.drawImage(Animation.getSyncFrame(ForestResources.hugoTraptalk, ForestResources.syncTrap, this.getFrameIndex()), 0, 0);
  }
}
